> lgtm. @christophebedard @MichaelOrlov can we merge this?

Unfortunately, not. The RPR job is failing with the error message:
```bash

2025-04-21T10:20:42.4834638Z 
2025-04-21T10:20:42.4835410Z /__w/rosbag2/rosbag2/ros_ws/src/fyxqkt4p44/rosbag2/rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp:203: error: ignoring ‘#pragma clang diagnostic’ [-Werror=unknown-pragmas]
2025-04-21T10:20:42.4836122Z   203 | #pragma clang diagnostic push
2025-04-21T10:20:42.4836336Z       | 
2025-04-21T10:20:42.4837132Z /__w/rosbag2/rosbag2/ros_ws/src/fyxqkt4p44/rosbag2/rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp:204: error: ignoring ‘#pragma clang diagnostic’ [-Werror=unknown-pragmas]
2025-04-21T10:20:42.4838283Z   204 | #pragma clang diagnostic ignored "-Wthread-safety-analysis"
2025-04-21T10:20:42.4838596Z       | 
2025-04-21T10:20:42.4839366Z /__w/rosbag2/rosbag2/ros_ws/src/fyxqkt4p44/rosbag2/rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp:206: error: ignoring ‘#pragma clang diagnostic’ [-Werror=unknown-pragmas]
2025-04-21T10:20:42.4840071Z   206 | #pragma clang diagnostic pop
2025-04-21T10:20:42.4840278Z       | 
2025-04-21T10:20:42.4840449Z cc1plus: all warnings being treated as errors
2025-04-21T10:20:42.4841003Z gmake[2]: *** [CMakeFiles/rosbag2_cpp.dir/build.make:146: CMakeFiles/rosbag2_cpp.dir/src/rosbag2_cpp/clocks/time_controller_clock.cpp.o] Error 1
2025-04-21T10:20:42.4841546Z gmake[2]: *** Waiting for unfinished jobs....
2025-04-21T10:20:42.4841893Z gmake[1]: *** [CMakeFiles/Makefile2:206: CMakeFiles/rosbag2_cpp.dir/all] Error 2
2025-04-21T10:20:42.4842246Z gmake[1]: *** Waiting for unfinished jobs....
2025-04-21T10:20:42.4842498Z gmake: *** [Makefile:146: all] Error 2
2025-04-21T10:20:42.4842708Z ---
2025-04-21T10:20:42.4842894Z Failed   <<< rosbag2_cpp [11.7s, exited with code 2]
```

Need to wrap clang pragma disabling in the 
```cpp
#if defined(__clang__)
// pargmas
#endif  // #if defined(__clang__)
```

--- a/rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp
+++ b/rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp
@@ -201,14 +201,14 @@ bool TimeControllerClock::is_sleeping()
 void TimeControllerClock::wakeup()
 {
 #if defined(__clang__)
-    #pragma clang diagnostic push
-    #pragma clang diagnostic ignored "-Wthread-safety-analysis"
+  #pragma clang diagnostic push
+  #pragma clang diagnostic ignored "-Wthread-safety-analysis"
 #endif
-    // Note. We shall not lock mutex before notify_all since the notified thread would immediately
-    // block again, waiting for the notifying thread to release the lock
-    impl_->cv.notify_all();
+  // Note. We shall not lock mutex before notify_all since the notified thread would immediately
+  // block again, waiting for the notifying thread to release the lock
+  impl_->cv.notify_all();
 #if defined(__clang__)
-    #pragma clang diagnostic pop
+  #pragma clang diagnostic pop
 #endif
 }
 

--- a/rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp
+++ b/rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp
@@ -200,10 +200,16 @@ bool TimeControllerClock::is_sleeping()
 
 void TimeControllerClock::wakeup()
 {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wthread-safety-analysis"
-  impl_->cv.notify_all();
-#pragma clang diagnostic pop
+#if defined(__clang__)
+    #pragma clang diagnostic push
+    #pragma clang diagnostic ignored "-Wthread-safety-analysis"
+#endif
+    // Note. We shall not lock mutex before notify_all since the notified thread would immediately
+    // block again, waiting for the notifying thread to release the lock
+    impl_->cv.notify_all();
+#if defined(__clang__)
+    #pragma clang diagnostic pop
+#endif
 }
 
 bool TimeControllerClock::set_rate(double rate)

